www/src/pages/foundations/elevation.jsx

```diff
@@ -0,0 +1,269 @@
+  const [boxShadowModel, setBoxShadowModel] = useState({
+    x: 10,
+    y: 10,
+    blur: 25,
+    spread: 0,
+    color: '#000',
+    inset: false,
+  });
+
+  const updateBoxShadowModel = (property, value) => {
+    setBoxShadowModel({
+      ...boxShadowModel,
+      [property]: value,
+    });
+  };
+
+  useEffect(() => {
+    updateBoxShadow(boxShadowModel, id);
+  }, [boxShadowModel]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  return (
+    <section className="pgn-doc__box-shadow-toolkit--controls-box">
+      {controlsProps.map(({ key, name }) => (
+        <Form.Label
+          className="d-block"
+          key={key}
+          htmlFor={`toolkit-control-${key}`}
+        >
+          {name}
+          <Input
+            id={`toolkit-control-${key}`}
+            key={key}
+            min={key === 'x' || key === 'y' ? '-100' : '0'}
+            max="100"
+            type={key === 'color' ? 'color' : 'range'}
+            defaultValue="0"
+            onChange={(e) => updateBoxShadowModel(key, e.target.value)}
+          />
+        </Form.Label>
+      ))}
+      <Form.Checkbox
+        onChange={() => updateBoxShadowModel('inset', !boxShadowModel.inset)}
+        floatLabelLeft
+      >
+        Inset:
+      </Form.Checkbox>
+    </section>
+  );
+};
+
+BoxShadowToolkit.propTypes = {
+  updateBoxShadow: PropTypes.func.isRequired,
+  id: PropTypes.number.isRequired,
+};
+
+const BoxShadowGenerator = () => {
+  const [boxShadows, setBoxShadows] = useState([]);
+  const [controls, setControls] = useState([{ index: 0 }]);
+
+  const updateBoxShadow = (shadow, id) => {
+    const boxShadow = [...boxShadows];
+    boxShadow[id] = shadow.inset ? `inset ${shadow.x}px ${shadow.y}px ${shadow.blur}px ${shadow.spread}px ${shadow.color}`
+      : `${shadow.x}px ${shadow.y}px ${shadow.blur}px ${shadow.spread}px ${shadow.color}`;
+    setBoxShadows(boxShadow);
+  };
+
+  const addBoxShadow = () => {
+    setControls([...controls, {
+      index: controls.length,
+    }]);
+  };
+
+  return (
+    <section className="pgn-doc__box-shadow-generator">
+      <div className="pgn-doc__box-shadow-generator--preview">
+        <div
+          className="pgn-doc__box-shadow-generator--preview-box border"
+          style={{ boxShadow: boxShadows.join(',') }}
+        />
+      </div>
+      <div className="pgn-doc__box-shadow-generator--toolkit">
+        <div className="d-flex overflow-auto mb-2">
+          {controls.map(control => (
+            <BoxShadowToolkit
+              key={control.index}
+              id={control.index}
+              updateBoxShadow={updateBoxShadow}
+            />
+          ))}
+        </div>
+        <div className="d-flex justify-content-between flex-column flex-md-row">
+          <code className="pgn-doc__box-shadow-generator--toolkit-code mb-3 mb-md-0">
+            box-shadow: {boxShadows.join(', ')};
+          </code>
+          <Button onClick={addBoxShadow} variant="dark">Add New Layer</Button>
```

Comment:
Can we add a Segment event here to track when users interact with this button? The event name might be `openedx.paragon.elevation.generator.layer.added`.

Reply:
Added